src/main/java/gui/viewmodel/TodoListPageViewModel.java

```diff
@@ -12,87 +13,78 @@
-            while (i < this.taskReaderList.size()) {
-                if (taskReaderList.get(i).getDeadline() != null) {
-                    if (deadline.isEqual(taskReaderList.get(i).getDeadline()) ||
-                            (deadline.isBefore(taskReaderList.get(i).getDeadline()))) {
-                        taskReaderList.add(i, taskReader);
+            while (i < this.taskInfoList.size()) {
+                if (taskInfoList.get(i).getDeadline() != null) {
+                    if (deadline.isEqual(taskInfoList.get(i).getDeadline()) ||
+                            (deadline.isBefore(taskInfoList.get(i).getDeadline()))) {
+                        taskInfoList.add(i, taskInfo);
                         return;
                     }
                 } else {
-                    taskReaderList.add(i, taskReader);
+                    taskInfoList.add(i, taskInfo);
                     return;
                 }
                 i = i + 1;
             }
         }
-        taskReaderList.add(taskReader);
+        taskInfoList.add(taskInfo);
     }
 
     /***
-     * Update TaskInfoList to make it consistent with taskReaderList
+     * Update viewInfoList to make it consistent with taskReaderList
      */
-    public void updateTaskInfoList() {
-        this.taskInfoList.clear();
-        for (TaskReader taskReader : this.taskReaderList) {
-            String id = String.valueOf(taskReader.getId());
-            String taskName = taskReader.getName();
+    public void updateViewInfoList() {
+        this.viewInfoList.clear();
+        for (TaskInfo taskInfo : taskInfoList) {
+            String id = String.valueOf(taskInfo.getId());
+            String taskName = taskInfo.getName();
 
             String deadline = new String("No Deadline");
-            if (taskReader.getDeadline() != null) {
-                deadline = taskReader.getDeadline().format(
+            if (taskInfo.getDeadline() != null) {
+                deadline = taskInfo.getDeadline().format(
                         DateTimeFormatter.ofLocalizedDateTime(
                                 FormatStyle.MEDIUM, // The format for date
                                 FormatStyle.SHORT)); // The format for time
             }
 
-            Map<String, String> taskInfo = new HashMap<String, String>();
-            taskInfo.put("id", id);
-            taskInfo.put("taskName", taskName);
-            taskInfo.put("deadline", deadline);
-            this.taskInfoList.add(taskInfo);
+            Map<String, String> taskViewInfo = new HashMap<String, String>();
+            taskViewInfo.put("id", id);
+            taskViewInfo.put("taskName", taskName);
+            taskViewInfo.put("deadline", deadline);
+            this.viewInfoList.add(taskViewInfo);
         }
     }
 
     public ObservableList<Map<String, String>> getTaskInfoList() {
-        return taskInfoList;
+        return viewInfoList;
     }
 
     public void handleCreation(TaskReader taskReader) {
```

Comment:
Still using `taskReader` here

Reply:
this task reader is received when the observable repo updates its observers. wasnt part of the refactoring.